cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+	Status string `"json:status,omitempty"`
+}
+
+type podPsOptions struct {
+	Filter             string
+	NoTrunc            bool
+	Format             string
+	Sort               string
+	Quiet              bool
+	NumberOfContainers bool
+	Cgroup             bool
+	Labels             bool
+	NamesOfContainers  bool
+	IdsOfContainers    bool
+	StatusOfContainers bool
+}
+
+type podPsTemplateParams struct {
+	ID                 string
+	Name               string
+	NumberOfContainers int
+	Status             string
+	Labels             string
+	Cgroup             string
+	UsePodCgroup       bool
+	ContainerInfo      string
+}
+
+// podPsJSONParams is used as a base structure for the psParams
+// If template output is requested, podPsJSONParams will be converted to
+// podPsTemplateParams.
+// podPsJSONParams will be populated by data from libpod.Container,
+// the members of the struct are the sama data types as their sources.
+type podPsJSONParams struct {
+	ID                 string         `json:"id"`
+	Name               string         `json:"name"`
+	NumberOfContainers int            `json:"numberofcontainers"`
+	Status             string         `json:"status"`
+	CtrsInfo           []podPsCtrInfo `json:"containerinfo,omitempty"`
+	Labels             fields.Set     `json:"labels,omitempty"`
+	Cgroup             string         `json:"cgroup,omitempty"`
+	UsePodCgroup       bool           `json:"podcgroup,omitempty"`
+}
+
+// Type declaration and functions for sorting the pod PS output
+type podPsSorted []podPsJSONParams
+
+func (a podPsSorted) Len() int      { return len(a) }
+func (a podPsSorted) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+type podPsSortedId struct{ podPsSorted }
+
+func (a podPsSortedId) Less(i, j int) bool { return a.podPsSorted[i].ID < a.podPsSorted[j].ID }
+
+type podPsSortedNumber struct{ podPsSorted }
+
+func (a podPsSortedNumber) Less(i, j int) bool {
+	return len(a.podPsSorted[i].CtrsInfo) < len(a.podPsSorted[j].CtrsInfo)
+}
+
+type podPsSortedName struct{ podPsSorted }
+
+func (a podPsSortedName) Less(i, j int) bool { return a.podPsSorted[i].Name < a.podPsSorted[j].Name }
+
+type podPsSortedStatus struct{ podPsSorted }
+
+func (a podPsSortedStatus) Less(i, j int) bool {
+	return a.podPsSorted[i].Status < a.podPsSorted[j].Status
+}
+
+var (
+	podPsFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "cgroup",
+			Usage: "Print the Cgroup information of the pod",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-names",
+			Usage: "Display the container names",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-ids",
+			Usage: "Display the container UUIDs. If no-trunc is not set they will be truncated",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-status",
+			Usage: "Display the container status",
+		},
+		cli.StringFlag{
+			Name:  "filter, f",
+			Usage: "Filter output based on conditions given",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "Pretty-print pods to JSON or using a Go template",
+		},
+		cli.BoolFlag{
```

Comment:
I don't think this one makes any sense, this is a potentially gigantic key-value dictionary. We shouldn't print it in `ps`, this is an `inspect` thing